src/auth/awss3/core/qgsauthawss3method.cpp

```diff
@@ -125,11 +125,13 @@ bool QgsAuthAwsS3Method::updateNetworkRequest( QNetworkRequest &request, const Q
                                 QCryptographicHash::Sha256 );
 
   const QByteArray signature = QMessageAuthenticationCode::hash( stringToSign, signingKey, QCryptographicHash::Sha256 ).toHex();
+  const QByteArray authorizationHeader = QString("%1 Credential=%2/%3/%4/s3/aws4_request, SignedHeaders=%5, Signature=%6")
+                                             .arg(encryptionMethod, username, date, region, headerList, signature)
+                                             .toLocal8Bit();
```

Comment:
Is it safe to use `toLocal8Bit()` here (and in some of the various other occurrences)? Shouldn't `toUtf8`() be used instead?

Reply:
yes toUtf8() seems far better/safer

Reply:
I updated all occurrences to use `toUtf8()`